MSAL/src/MSALErrorConverter.m

```diff
@@ -135,14 +135,25 @@ + (NSError *)msalErrorFromMsidError:(NSError *)msidError
 + (NSError *)msalErrorFromMsidError:(NSError *)msidError
                      classifyErrors:(BOOL)shouldClassifyErrors
                  msalOauth2Provider:(MSALOauth2Provider *)oauth2Provider
+{
+    return [self msalErrorFromMsidError:msidError
+                         classifyErrors:shouldClassifyErrors
+                     msalOauth2Provider:oauth2Provider
+                          correlationId:nil];
+}
+
++ (NSError *)msalErrorFromMsidError:(NSError *)msidError
+                     classifyErrors:(BOOL)shouldClassifyErrors
+                 msalOauth2Provider:(MSALOauth2Provider *)oauth2Provider
+                            correlationId:(NSUUID *)correlationId
 {
     return [self errorWithDomain:msidError.domain
                             code:msidError.code
                 errorDescription:msidError.userInfo[MSIDErrorDescriptionKey]
                       oauthError:msidError.userInfo[MSIDOAuthErrorKey]
                         subError:msidError.userInfo[MSIDOAuthSubErrorKey]
                  underlyingError:msidError.userInfo[NSUnderlyingErrorKey]
-                   correlationId:msidError.userInfo[MSIDCorrelationIdKey]
+                   correlationId:msidError.userInfo[MSIDCorrelationIdKey] ? : correlationId
```

Comment:
is msidError.userInfo[MSIDCorrelationIdKey] also NSUUID type? Make sure both options are of the same type.